generators/csharp/sdk/src/test-generation/mock-server/MockServerTestGenerator.ts

```diff
@@ -135,29 +129,44 @@ export class MockServerTestGenerator extends FileGenerator<CSharpFile, SdkCustom
-                    writer.write("Assert.DoesNotThrowAsync(async () => ");
-                }
-                writer.writeNode(endpointSnippet);
-                if (!responseSupported) {
-                    writer.write(")");
-                }
-                writer.write(";");
+                if (this.endpoint.pagination) {
+                    writer.write("var pager = ");
+                    writer.writeNode(endpointSnippet);
+                    writer.write(";");
+                    writer.newLine();
+                    writer.write("await foreach (var item in pager)");
+                    writer.newLine();
+                    writer.write("{");
+                    writer.newLine();
+                    writer.indent();
 
-                if (responseSupported) {
+                    writer.writeTextStatement("Assert.That(item, Is.Not.Null)");
+                    writer.write("break; // Only check the first item");
+
+                    writer.dedent();
                     writer.newLine();
-                    if (responseBodyType === "json") {
-                        writer.addReference(this.context.getFluentAssetionsJsonClassReference());
-                        writer.writeNode(this.context.getJTokenClassReference());
-                        writer.write(".Parse(mockResponse).Should().BeEquivalentTo(");
-                        writer.writeNode(this.context.getJTokenClassReference());
-                        writer.write(".Parse(");
-                        writer.writeNode(this.context.getJsonUtilsClassReference());
-                        writer.writeTextStatement(".Serialize(response)))");
-                    } else if (responseBodyType === "text") {
-                        writer.writeTextStatement("Assert.That(response, Is.EqualTo(mockResponse))");
+                    writer.write("}");
+                } else {
+                    if (responseSupported) {
+                        writer.write("var response = ");
+                        writer.writeNode(endpointSnippet);
+                        writer.write(";");
+                        writer.newLine();
+                        if (responseBodyType === "json") {
+                            writer.addReference(this.context.getFluentAssetionsJsonClassReference());
+                            writer.writeNode(this.context.getJTokenClassReference());
+                            writer.write(".Parse(mockResponse).Should().BeEquivalentTo(");
+                            writer.writeNode(this.context.getJTokenClassReference());
+                            writer.write(".Parse(");
+                            writer.writeNode(this.context.getJsonUtilsClassReference());
+                            writer.writeTextStatement(".Serialize(response)))");
+                        } else if (responseBodyType === "text") {
+                            writer.writeTextStatement("Assert.That(response, Is.EqualTo(mockResponse))");
+                        }
```

Comment:
Let's use `switch` w/ `assertNever` - that way we get compile breaks if we ever add more response body types.

When we have the switch, it might be worth rolling out helper functions for these cases (but I'll leave it to you to decide).